scripts/format_translations.py

```diff
@@ -201,18 +195,19 @@ def main():
                 sys.exit()
 
     with open(args.filename, 'rb') as csvfile:
-        reader = csv.reader(csvfile)
-        for row in reader:
-            keys = row[0].split('.')
-            merge(data, format_dict(keys, row[1].strip(" ")))
+            reader = csv.reader(csvfile)
+            for row in reader:
+                keys = row[0].split('.')
+                merge(data, format_dict(keys, row[1].strip(" ")))
+
     with open(out_fn, 'w') as f:
-        data.update(numbers)
-        json.dump(data, f, indent=4, sort_keys=True, ensure_ascii=False)
+            data.update(numbers)
+            json.dump(data, f, indent=4, sort_keys=True, ensure_ascii=False)
 
-    if args.validate:
-        with open(REFERENCE_LOCALE_PATH, 'r') as f:
+    with open(REFERENCE_LOCALE_PATH, 'r') as f:
             reference_translation = json.load(f)
-        validate_translations(reference_translation, data)
+            validate_translations(reference_translation, data)
```

Comment:
As discussed, it would be better to validate *before* writing any output file, and stop the script using `sys.exit(1)` if validation fails. 

Also, this change seems to remove the check for the validation flag entirely- validation will always happen no matter what option the user provides. 

I'm actually ok with making validation mandatory... but if you remove the behavior controlled by this flag, then I would suggest updating the usage example (line 9) and removing the argument from argparse (line 179).  Otherwise it's just confusing.